strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/PythonWheelArtifactGenerator.java

```diff
@@ -30,24 +34,59 @@
                                                    "\n" +
                                                    "Strongbox wheel package for test";
 
-    private String basedir;
+    private Path basedir;
+    
+    
+    public PythonWheelArtifactGenerator(Path basedir)
+    {
+        this.basedir = basedir;
+    }
 
     public PythonWheelArtifactGenerator(String basedir)
     {
-        this.basedir = basedir;
+        this.basedir = Paths.get(basedir);
+    }
+
+    @Override
+    public Path generateArtifact(String id,
+                                 String version,
+                                 int size)
+            throws IOException
+    {
+        PypiWheelArtifactCoordinates coordinates = new PypiWheelArtifactCoordinates(id,
+                                                                                    version,
+                                                                                    null,
+                                                                                    "py3",
+                                                                                    "none",
+                                                                                    "any");
+        return generateArtifact(coordinates);
     }
 
-    public void generateWheelPackage(PypiWheelArtifactCoordinates coordinates)
+    @Override
+    public Path generateArtifact(URI uri,
+                                 int size)
             throws IOException
     {
-        String packagePath = String.format("%s/%s-%s-py2-none-any.whl", basedir, coordinates.getId(),
+        PypiWheelArtifactCoordinates coordinates = PypiWheelArtifactCoordinates.parse(FilenameUtils.getName(uri.getPath()));
+        return generateArtifact(coordinates);
+    }
+
+    public Path generateArtifact(PypiWheelArtifactCoordinates coordinates)
+            throws IOException
+    {
+        String packagePath = String.format("%s-%s-py2-none-any.whl",
+                                           coordinates.getId(),
                                            coordinates.getVersion());
 
-        ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(packagePath));
+        Path fullPath = basedir.resolve(packagePath);
+
+        ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(fullPath.toString()));
```

Comment:
@Firstmol you can use `Files.newOutputStream(fullPath)`

Reply:
ok

Reply:
fixed

Reply:
One more thing, can you please use `try-with-resources` statement for `OutputStream` here.